charts/redpanda/templates/secrets.yaml

```diff
@@ -136,3 +136,119 @@ stringData:
+      
+      process_users() {
+        USERS_DIR=${1-"/etc/secrets/users"}
+        USERS_FILE=$(find ${USERS_DIR}/* -print)
+        USERS_LIST=""
+        READ_LIST_SUCCESS=0
+        while read p; do
+          IFS=":" read -r USER_NAME PASSWORD MECHANISM <<< $p
+          # Do not process empty lines
+          if [ -z "$USER_NAME" ]; then
+            continue
+          fi
+          echo "Creating user ${USER_NAME}..."
+          MECHANISM=${MECHANISM:-{{- include "sasl-mechanism" . }}}
+          creation_result=$(rpk acl user create ${USER_NAME} --password=${PASSWORD} --mechanism ${MECHANISM} {{ template "rpk-flags-no-sasl" $ }} 2>&1) && creation_result_exit_code=$? || creation_result_exit_code=$?  # On a non-success exit code
+          if [[ $creation_result_exit_code -ne 0 ]]; then
+            # Check if the stderr contains "User already exists"
+            # this error occurs when password has changed
+            if [[ $creation_result == *"User already exists"* ]]; then
+              echo "Update user ${USER_NAME}"
+              # we will try to update by first deleting
+              deletion_result=$(rpk acl user delete ${USER_NAME} {{ template "rpk-flags-no-sasl" $ }} 2>&1) && deletion_result_exit_code=$? || deletion_result_exit_code=$?
+              if [[ $deletion_result_exit_code -ne 0 ]]; then
+                echo "deletion of user ${USER_NAME} failed: ${deletion_result}"
+                READ_LIST_SUCCESS=1
+                break
+              fi
+              # Now we update the user
+              update_result=$(rpk acl user create ${USER_NAME} --password=${PASSWORD} --mechanism ${MECHANISM} {{ template "rpk-flags-no-sasl" $ }} 2>&1) && update_result_exit_code=$? || update_result_exit_code=$?  # On a non-success exit code
+              if [[ $update_result_exit_code -ne 0 ]]; then
+                echo "updating user ${USER_NAME} failed: ${update_result}"
+                READ_LIST_SUCCESS=1
+                break
+              else
+                echo "Updated user ${USER_NAME}..."
+                USERS_LIST=$(new_users_list "${USERS_LIST}" "${USER_NAME}")
+              fi
+            else
+              # Another error occurred, so output the original message and exit code
+              echo "error creating user ${USER_NAME}: ${creation_result}"
+              READ_LIST_SUCCESS=1
+              break
+            fi
+          # On a success, the user was created so output that
+          else
+            echo "Created user ${USER_NAME}..."
+            USERS_LIST=$(new_users_list "${USERS_LIST}" "${USER_NAME}")
+          fi
+        done < $USERS_FILE
+            
+        if [[ -n "${USERS_LIST}" && ${READ_LIST_SUCCESS} ]]; then
+          echo "Setting superusers configurations with users [${USERS_LIST}]"
+          superuser_result=$(rpk cluster config set superusers [${USERS_LIST}] {{ template "rpk-flags-no-sasl" $ }} 2>&1) && superuser_result_exit_code=$? || superuser_result_exit_code=$?
```

Comment:
Isn't helm chart suppose to manage `redpanda.yaml` file with the super users list? 

https://github.com/redpanda-data/helm-charts/blob/9b20abf81e80b2f265a3e75335c002d2ea9cf5c7/charts/redpanda/templates/configmap.yaml#L70

https://github.com/redpanda-data/helm-charts/blob/9b20abf81e80b2f265a3e75335c002d2ea9cf5c7/charts/redpanda/templates/configmap.yaml#L114

Reply:
Only for bootstraping and only if the userlist is not empty. In the case you are providing your own user list this will not be accepted. In the case of updating list, this list will be out of date.